src/models/TranslationValue/addTranslationValueToTranslationKey/index.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import TranslationValue from '@models/TranslationValue'
+import TranslationKey from '@models/TranslationKey'
+import Locale from '@models/Locale'
+import type { ValidationError } from '@types/ValidationError'
+
+export type AddTranslationValueToTranslationKeyArgs = {
+  value: string,
+  translationKeyId: string,
+  localeId: string,
+  ownerId: string
+}
+
+type AddTranslationValueToTranslationKeyResponse = {
+  translationKey?: TranslationKey,
+  errors: Array<ValidationError>
+}
+
+const addTranslationValueToTranslationKey = async ({
+  value,
+  translationKeyId,
+  localeId,
+  ownerId
+}: AddTranslationValueToTranslationKeyArgs): Promise<
+  AddTranslationValueToTranslationKeyResponse
+> => {
+  const errors: Array<ValidationError> = []
+
+  const translationKey: TranslationKey = await TranslationKey.query()
+    .findById(translationKeyId)
+    .eager('[project.locales, translationValues.locale]')
+
+  if (!translationKey) {
+    throw new Error('This translation key was not found.')
+  }
+
+  const project = translationKey.project
+  const locale = await Locale.query().findById(localeId)
+
+  if ((project && project.ownerId !== ownerId) || !locale) {
+    throw new Error('This project was not found.')
+  }
+
+  let isLocaleActivated = false
+  project.locales.forEach(projectLocale => {
+    if (projectLocale.id === locale.id) {
+      isLocaleActivated = true
+    }
+  })
```

Comment:
Same here, if a project had hundreds of locales, it could impact performance when it gets parsed to a JavaScript object.
We could probably have a query for that instead…

If wwe end up keeping that, let‘s rewrite it to:
```js
const isLocaleActivated = !!project.locales.find(projectLocale => projectLocale.id === locale.id)
```

The `!!` makes it a boolean. You could also use `Boolean(…)`.

Reply:
Can we do that ☝️ in case we won’t refactor it later?